api/src/main/java/org/openmrs/Condition.java

```diff
@@ -105,8 +105,8 @@ public Condition(CodedOrFreeText condition, ConditionClinicalStatus clinicalStat
 		this.verificationStatus = verificationStatus;
 		this.previousVersion = previousVersion;
 		this.additionalDetail = additionalDetail;
-		this.onsetDate = onsetDate;
-		this.endDate = endDate;
+		this.onsetDate = onsetDate != null ? (Date) onsetDate.clone() : null;
```

Comment:
Did you get a chance to look at this? http://www.informit.com/articles/article.aspx?p=31551&seqNum=2

Reply:
Well, in my previous commit I used `new Data(onsetDate.getTime())` but suthagar reccommended me to maintain consistency between `new Data(onsetDate.getTime())` and `(Date) onsetDate.clone()` so I choosed the second one as the first method didn't work when returning the value

Reply:
After reading that page, what do you think is the recommended way forward?

Reply:
The artical does not recommend to use the `clone()` method as it posses some security risks, I think we should go for `new Date(onsetDate.getTime())`

Reply:
Does it not recommend clone in all cases? Or just some?

Reply:
`clone()` should be used only when returning values and new copy must be made while setting the values, so in my case I'll have to replace the clone with new copy in setters

Reply:
I'll push those changes soon @dkayiwa